src/core/src/sys/molecules.rs

```diff
@@ -176,39 +48,39 @@ impl Molecule {
             angles: HashSet::new(),
             dihedrals: HashSet::new(),
             connections: Array2::default((1, 1)),
-            first: i,
-            last: i,
+            atoms: i..i+1,
             cached_hash: 0
         }
     }
 
     /// Get the number of atoms in the molecule
     pub fn size(&self) -> usize {
-        self.last - self.first + 1
+        self.atoms.len()
     }
 
     /// Get the first atom of this molecule
     pub fn first(&self) -> usize {
-        self.first
+        self.atoms.start
     }
 
-    /// Get the last atom of this molecule
-    pub fn last(&self) -> usize {
-        self.last
+    /// Get the index of the first atom after this molecule
+    pub fn next_atom(&self) -> usize {
```

Comment:
How about `end` then, like the std `Range`? The point is that this value is the first index not in the molecule.